File: array/0832_flipping_an_image/0832_flipping_an_image.py
from typing import List

class Solution:
    def flipAndInvertImage(self, A: List[List[int]]) -> List[List[int]]:
        m , n = len(A), len(A[0])
        for i in range(m):
            j, k = 0, n -1
            while j < k:
                A[i][j], A[i][k] = A[i][k], A[i][j]
                j += 1
                k -= 1
        
        for i in range(m):
            for j in range(n):
                A[i][j] = 1 if A[i][j] == 0 else 0
        
        return A

File: array/0832_flipping_an_image/0832_flipping_an_image.go
package main

import "fmt"

func flipAndInvertImage(A [][]int) [][]int {
	for i, row := range A {
		for j := range row {
			if j > (len(row)-1)/2 {
				break
			}

			tmp := A[i][j]
			A[i][j] = A[i][len(row)-1-j]
			A[i][len(row)-1-j] = tmp

			A[i][j] = A[i][j] ^ 1
			if len(row)-1-j != j {
				A[i][len(row)-1-j] = A[i][len(row)-1-j] ^ 1

			}
		}
	}
	return A
}

func main() {
	A := [][]int{{1, 1, 0}, {1, 0, 1}, {0, 0, 0}}
	res := flipAndInvertImage(A)
	fmt.Println(res)
}
